core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -149,16 +148,32 @@ public void onComplete() {
      */
     @Override
     public boolean tryComplete() {
-        topicPartitionDataFromTryComplete = acquirablePartitions();
-
-        if (!topicPartitionDataFromTryComplete.isEmpty()) {
-            boolean completedByMe = forceComplete();
-            // If invocation of forceComplete is not successful, then that means the request is already completed
-            // hence release the acquired locks.
-            if (!completedByMe) {
-                releasePartitionLocks(shareFetchData.groupId(), topicPartitionDataFromTryComplete.keySet());
+        Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData = acquirablePartitions();
+
+        if (!topicPartitionData.isEmpty()) {
+            FetchOffsetMetadataUpdateResult fetchOffsetMetadataUpdateResult = maybeUpdateFetchOffsetMetadataForTopicPartitions(topicPartitionData);
+            if (isMinBytesSatisfied(topicPartitionData)) {
+                topicPartitionDataFromTryComplete = topicPartitionData;
+                if (fetchOffsetMetadataUpdateResult.isFetchOffsetMetadataUpdated
```

Comment:
This condition seems unnecessary. We need to set `logReadResponse` as long as `fetchOffsetMetadataUpdateResult.replicaManagerReadResponse` is not empty, right?